pkg/collector/reconcile/config_replace.go

```diff
@@ -71,13 +87,16 @@ func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
 		}
 	}
 
-	updPromCfgMap := make(map[string]interface{})
-	if err := mapstructure.Decode(cfg, &updPromCfgMap); err != nil {
-		return "", err
+	if EnableTargetAllocatorRewrite.IsEnabled() {
+		cfg.TargetAllocConfig = &targetAllocator{
+			Endpoint:    fmt.Sprintf("http://%s:80", naming.TAService(instance)),
+			Interval:    30 * time.Second,
```

Comment:
If someone sets these values already in the Spec that is recovered from the call to `ConfigToPromConfig` will this overwrite those values?

Reply:
yes, in this case we will overwrite whatever a user may have set.

Reply:
Oh so when the featuregate is disabled then the user settings is not overwritten? Or is the user not really supposed to set this part of the config?

Reply:
yeah when the featuregate is disabled, nothing will happen which means whatever settings are there will remain.

Reply:
How should this be handled when the feature gate becomes beta or stable and is enabled by default?  Should there be a log entry indicating the user-provided content is being replaced?

Reply:
yeah that seems reasonable, FWIW we already replace user content for this functionality above for the http_sd_config destination. That will be in a follow up when this is moved to beta